src/backend/app/app.py

```diff
@@ -25,6 +38,48 @@ def create_table():
             )
         ''')
 
+
+create_table()
+
+def allowed_file(filename):
+    """Check if the uploaded file is allowed."""
+    return '.' in filename and filename.rsplit('.', 1)[1].lower() in app.config['ALLOWED_EXTENSIONS']
+
+def extract_text_from_docx(file_path):
+    """Extract text from a .docx file."""
+    doc = docx.Document(file_path)
+    return "\n".join(paragraph.text for paragraph in doc.paragraphs)
+
+
+def extract_keywords(text):
```

Comment:
The list of potential keywords looks effective and diverse, good work

Reply:
Thanks!